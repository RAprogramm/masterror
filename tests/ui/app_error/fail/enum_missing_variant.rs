// SPDX-FileCopyrightText: 2025 RAprogramm <andrey.rozanov.vl@gmail.com>
//
// SPDX-License-Identifier: MIT

use masterror::{AppErrorKind, Error};

#[derive(Debug, Error)]
enum Mixed {
    #[error("with spec")]
    #[app_error(kind = AppErrorKind::NotFound)]
    WithSpec,
    #[error("without")]
    Without,
}

fn main() {}
