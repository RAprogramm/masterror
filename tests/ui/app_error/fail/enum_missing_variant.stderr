error: all variants must use #[app_error(...)] to derive AppError conversion
 --> tests/ui/app_error/fail/enum_missing_variant.rs:8:5
  |
8 | /     #[error("without")]
9 | |     Without,
  | |___________^

warning: unused import: `AppErrorKind`
 --> tests/ui/app_error/fail/enum_missing_variant.rs:1:17
  |
1 | use masterror::{AppErrorKind, Error};
  |                 ^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` (part of `#[warn(unused)]`) on by default
